### DISCUSSION TITLE
"GDB OpenOCD Debugging" Missing from v4.4.3?

### DISCUSSION THREAD

**Discussion by RatGit** [Sloeber/arduino-eclipse-plugin#1642] [Q&A] [answered]
Previously in v4.3.0, I used the "GDB OpenOCD Debugging" Debug Configuration for hardware debugging of my custom SAMD21G18A board. This option is now missing in v4.4.3?

Can anyone please explain how I can get this functionality back?

**Comment by jantje** [comment, marked as answer]
the  "GDB OpenOCD Debugging" is not a part of Sloeber but a part of the ecosystem of CDT.
This specific part moved closer to cdt and is now called embedded CDT. 
Due to all the renaming I'm not sure which part exactly you need to install but installing embedded CDT in your sloeber should provide you with the same (or upgraded) functionality.

**Reply by RatGit** [reply]
Thanks Jantje... BTW this is Keith, (previously from SmartaIndustrial). I've moved to a new company and have just introduced them to Sloeber! :)

**Reply by jantje** [reply]
Lol I'm still working on the lib stuff.
Good look on the new job.

**Reply by RatGit** [reply]
Great... I'd be happy to beta test any library compiler support whenever you're ready.

**Comment by RatGit** [comment]
Here is a little more procedural detail for others who might wish to do this:

**In order to get the "GDB OpenOCD Debugging" working in Sloeber v 4.4.3:**

**1. Install "Embedded CDT OpenOCD Debugging Plugin:**

Help > Install New Software... > Add...
   Eclipse Embedded CDT Plug-ins - https://download.eclipse.org/embed-cdt/updates/neon/

Install:
   Embedded C/C++ Cross Development Tools (incubation) > Embedded C/C++ OpenOCD Debugging (incubation)

**2. Create a "Debug" Build Configuration by copying "Release"**

**3. Modify the following Build Variables for the "Debug" Build Configuration:**
   Right-Click project and select: Properties > C/C++ Build > Build Variables

Modify ${compiler.c.elf.flags} by replacing  ${compiler.optimization_flags} with ${compiler.optimization_flags.debug}
Modify ${compiler.c.flags} by replacing  ${compiler.optimization_flags} with ${compiler.optimization_flags.debug}
Modify ${compiler.cpp.flags} by replacing  ${compiler.optimization_flags} with ${compiler.optimization_flags.debug}

**Reply by jantje** [reply]
Thanks for the writeup
